allennlp/models/bidirectional_lm.py

```diff
@@ -227,10 +236,11 @@ def forward(self,  # type: ignore
         forward_targets[:, 0:-1] = token_ids[:, 1:]
         backward_targets[:, 1:] = token_ids[:, 0:-1]
 
+        # shape (batch_size, timesteps + 2, embedding_size)
         embeddings = self._text_field_embedder(source)
 
         # Apply LayerNorm if appropriate.
-        embeddings = self._layer_norm(embeddings)
+        embeddings = self._layer_norm(embeddings, mask)
```

Comment:
Just to double check, did you mean to add layer norm both here and at the output of the `HighwayCnnEncoder`? I can see that might be useful for someone, but you probably don't want both these layernorms happening simultaneously.

Reply:
Ok I see why it's in both places now - you might not use a char cnn, but you might still want `LayerNorm` over your token embeddings. Please could you add to the docstring of the constructor that basically you should leave `do_layer_norm=False` if you are using the `char_cnn_encoder` with the default options?

Yes, it seems that that MaskedLayerNorm definition is actually doing `MaskedBatchNormalisation`. I'll defer to Matt as he wrote that originally in Calypso but it is equally "valid".

Reply:
Agree with @DeNeutoy that we shouldn't apply normalization twice.  @brendan-ai2 your proposal sounds fine to me, this would also remove the `do_layer_norm` parameter from this class and some duplicate code.  Is this the scenario you have been training?